### PR TITLE
Add array_key_first and array_key_last functions.

### LINKED ISSUE #348: RFC: array_key_first(), array_key_last()
https://wiki.php.net/rfc/array_key_first_last

### PR DESCRIPTION
**Pull request by nstdio**
Closes #348.

### CODE REVIEW

File: exts/jphp-zend-ext/src/main/java/org/develnext/jphp/zend/ext/standard/ArrayFunctions.java [hunk 1]
```diff
@@ -586,6 +587,23 @@ public static Memory array_keys(Environment env, TraceInfo trace, Memory input,
...
+        return array_key_impl(input, false);
+    }
+
+    private static Memory array_key_impl(Memory input, boolean first) {
+        if (!input.isArray() || empty(input)) {
+            return Memory.NULL;
+        }
+
+        ForeachIterator it = input.toValue(ArrayMemory.class).foreachIterator(false, false);
+        return (first ? it.next() : it.end()) ? it.getMemoryKey() : Memory.NULL;
```
Comment (dim-s) [inline exts/jphp-zend-ext/src/main/java/org/develnext/jphp/zend/ext/standard/ArrayFunctions.java]: `it.end()` will iterate all items in array, see ItemsUtils class and its `last` and `lastKey` implementation.

Comment (dim-s) [reply exts/jphp-zend-ext/src/main/java/org/develnext/jphp/zend/ext/standard/ArrayFunctions.java]: https://github.com/jphp-group/jphp/blob/d9b269526affe5d0dd763b53fcc7ba2bd07f6738/jphp-runtime/src/php/runtime/ext/core/classes/lib/ItemsUtils.java#L372

Comment (nstdio) [reply exts/jphp-zend-ext/src/main/java/org/develnext/jphp/zend/ext/standard/ArrayFunctions.java]: @dim-s 

After changes made recently `it.getMemoryKey()` does not work properly.

```php
$array = array( "1" => "one", "2" => "two", "3" => "three", "4" => "four", "5" => "five" );
var_dump(array_key_first($array));
```
Expected: `int(1)`.
Actual: `string(1) "1"`.

Can you take a look?

Comment (dim-s) [reply exts/jphp-zend-ext/src/main/java/org/develnext/jphp/zend/ext/standard/ArrayFunctions.java]: @nstdio I've fixed: https://github.com/jphp-group/jphp/commit/1fc9bed61e984180a9a76511bde95bdf3360d243

**Pull request merged**